src/components/network/layers/arrow-layer.js

```diff
@@ -206,43 +206,29 @@ export default class ArrowLayer extends Layer {
             const linePositionsTextureOffset = linePositionsTextureData.length / 2;
             const lineDistancesTextureOffset = lineDistancesTextureData.length;
             let linePointCount = 0;
-            let lineDistance = 0;
             if (positions.length > 0) {
-                let ruler = cheapRuler(positions[0][1], 'meters');
-                let prevPosition;
-                positions.forEach((position, index) => {
-                    // compute distance with previous position
-                    let segmentDistance = 0;
-                    if (prevPosition) {
-                        ruler = cheapRuler(position[1], 'meters');
-                        segmentDistance = ruler.distance([prevPosition[1], prevPosition[0]],
-                                                         [position[1], position[0]]);
-                    }
-                    prevPosition = position;
-
+                positions.forEach((position) => {
                     // fill line positions texture
                     linePositionsTextureData.push(position[0]);
                     linePositionsTextureData.push(position[1]);
-
                     linePointCount++;
-
-                    // fill line distance texture
-                    lineDistance += segmentDistance;
-                    lineDistancesTextureData.push(lineDistance);
                 });
+                lineDistancesTextureData.push(...line.cumulativeDistances);
             }
             if (linePointCount > MAX_LINE_POINT_COUNT) {
                 throw new Error(`Too many line point count (${linePointCount}), maximum is ${MAX_LINE_POINT_COUNT}`);
             }
-            lineAttributes.set(line, {
-                distance: lineDistance,
-                positionsTextureOffset: linePositionsTextureOffset,
-                distancesTextureOffset: lineDistancesTextureOffset,
-                pointCount: linePointCount
-            });
+            if (positions.length > 0) {
```

Comment:
Is it correct to not set the attributes at all ? Before this change it was set. I think It's easier to get right if we keep the previous behavior.